drivers/Aqara/aqara-lock/src/credential_utils.lua

```diff
@@ -0,0 +1,134 @@
+local capabilities = require "st.capabilities"
+local utils = require "st.utils"
+
+local remoteControlStatus = capabilities.remoteControlStatus
+local lockCredentialInfo = capabilities["stse.lockCredentialInfo"]
+
+local credential_utils = {}
+local HOST_COUNT = "__host_count"
+
+credential_utils.save_data = function(driver)
+  driver.datastore:save()
+  driver.datastore:commit()
+end
+
+credential_utils.update_remote_control_status = function(driver, device, added)
+  local host_cnt = device:get_field(HOST_COUNT) or 0
+  if added then
+    if host_cnt == 0 then
+      device:emit_event(remoteControlStatus.remoteControlEnabled('true', { visibility = { displayed = false } }))
+    end
+    host_cnt = host_cnt + 1
+  else
+    if host_cnt > 0 then host_cnt = host_cnt - 1 end
+
+    if host_cnt == 0 then
+      device:emit_event(remoteControlStatus.remoteControlEnabled('false', { visibility = { displayed = false } }))
+    end
```

Comment:
if the intention is for this to only fire the _first_ time the `host_cnt` goes below 0, you may want to move it inside the if statement above. Otherwise you may still have an issue if there is a remove event when the value is 0.

Reply:
Which issue do you mean? as far as i know, `host_cnt` and `remoteControlStatus.remoteControlEnabled` are using for displaying on/off button on device card. My intention is that if there is no "host" credentionInfo list, edge driver send `remoteControlStatus.remoteControlEnabled` with false.

Reply:
I guess it doesn't matter because it will be debounced as a duplicate anyway.